enhancements/vsphere-multi-vcenter.md

```diff
@@ -0,0 +1,839 @@
+our needs for High Availability.
+
+As a system administrator, I would like to scale new nodes across multiple 
+vSphere vCenters so that I can leverage various availability zones for 
+workloads depending on our organization's needs.
+
+As a system administrator, I would like to add a new vCenter to the existing
+OCP cluster so that I can scale out new workloads across a new vCenter.
+
+### Goals
+```
+Summarize the specific goals of the proposal. How will we know that
+this has succeeded?  A good goal describes something a user wants from
+their perspective, and does not include the implementation details
+from the proposal.
+```
+
+- During installation, all nodes are created in all defined vCenters.
+  Rational: multiple vCenters is another twist on failure domains.  A new vCenter 
+  will appear as a new failure domain to the OCP cluster.  Control plane and 
+  compute nodes must be able to be assigned to any FD.
+
+- OCP clusters will be enhanced to leverage new yaml cloud provider config for
+  vSphere.
+  Rational: The ini configuration has been deprecated and the newer yaml format
+  supports multiple vCenters.
+
+- When updating infrastructure after initial installation, the cluster should be
+  able to accept the newly defined failure domain which points leverages a new 
+  vCenter.  
+
+- Updating the cloud provider config from ini to yaml will be supported.
+  Rational: Existing clusters wish to take advantage of migrating loads to a new
+  vCenter.  In order for this to happen, we must be able to allow customers to
+  update their existing cloud provider config to contain all relevant information
+  for the new vCenter.
+
+### Non-Goals
+
+What is out of scope for this proposal? Listing non-goals helps to
+focus discussion and make progress. Highlight anything that is being
+deferred to a later phase of implementation that may call for its own
+enhancement.
+
+- Updating cloud config to yaml format for existing clusters (upgrading OCP to 4.17+)
+
+## Proposal
+
+### CRD Changes
+
+The multiple vCenter feature will begin allowing more than one vCenter to be 
+configured in the infrastructure resource.  We will be controlling this via
+a new feature (VSphereMultiVCenters) and will have different CRDs installed
+based on this gate.
+
+Initially, the plans are to allow a max of 3 vCenters to be configured when the
+feature gate is enabled.  The way we are going to control this is by adding new
+control annotations to the model objects.
+
+The OpenShift controller tools will be enhanced to allow a new Feature Gate 
+Aware config option for max size.
+
+Example:
+```go
+// +kubebuilder:validation:MinItems=0
+// +openshift:validation:FeatureGateAwareMaxItems:featureGate="",maxItems=1
+// +openshift:validation:FeatureGateAwareMaxItems:featureGate=VSphereMultiVCenters,maxItems=3
+// +listType=atomic
+// +optional
+VCenters []VSpherePlatformVCenterSpec `json:"vcenters,omitempty"`
```

Comment:
Should we consider a webhook to block updates if the cluster doesnt have a yaml cloud config?

Reply:
We recently updated the CRD to block changes once this field is set in 4.17.  Once we finish migration testing and fixes, the CRD will be updated to not have those blockers anymore.  Right now I do believe the cluster admin will have to manually migrate this config.  I am hoping we can maybe create an automated process for this.